### DISCUSSION TITLE
Fire event from backend to frontend

### DISCUSSION THREAD

**Discussion by dmm9** [eclipse-theia/theia#13856] [Q&A]
I am trying to send an event from the backend to the frontend, and I don't know how. The use case: I have an HTTP endpoint on the backend, and I want to pass the data received on this endpoint to the frontend. RPC seems to be triggered always by the frontend. And Emitters seem not to jump between server and client. Am I mistaken? How can this be realized?

**Comment by msujew** [comment]
Hey @dmm9,

this is possible, but non-trivial. We use such a mechanism when sending file system events to the frontend. See [here](https://github.com/eclipse-theia/theia/blob/master/packages/filesystem/src/common/filesystem-watcher-protocol.ts) for some code. You can probably adapt it to your use case.

Note that Theia doesn't have "the" frontend - a backend can have a number of concurrently connected frontends. I.e. in Electron (the same is true for the browser), any single window still talks to the same backend. It's useful to keep that in mind when dealing with data sent from the backend to a frontend.

**Reply by dmm9** [reply]
The filesystem package is huge. Could you point me to a more specific function? Or maybe describe briefly what mechanism is used to do backend -> frontend communication?

**Reply by msujew** [reply]
I've pointed you towards the specific interfaces that are responsible for transmitting the file system change notifications to the frontend. I've also mentioned that it's non-trivial, given that it's just not one specific function but rather a whole RPC architecture at play.

For general backend to frontend communication, take a look at [this file](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/node/messaging/connection-container-module.ts), which also contains a bit of documentation on how to use it.

**Reply by dmm9** [reply]
Thanks @msujew , I will have a deeper look

**Reply by Yahud100** [reply]
Hii @dmm9  did you have progress on this ?

**Reply by dmm9** [reply]
Hi @Yahud100 , unfortunately not. I have to go with promises and polling from client side.
